tools/client-plugins/browser-scripts/frame-runner.ts

```diff
@@ -92,7 +92,13 @@ async function initTestFrame(e: InitTestFrameArg = { code: {} }) {
         // frameDocument ready:
         $(() => {
           try {
-            const test: unknown = eval(testString);
+            let test: unknown;
+            if (__file('script.js').indexOf(editableContents) > -1) {
+              const script = editableContents;
+              test = eval(script + ';' + testString);
```

Comment:
Have you investigated using `new Function(…)` instead? It may allow some better context isolation and possibly better performance than `eval`.

One thing to consider: Can I pass any exercise by setting `editableContents` to `'resolve(0)'`? 😈

Reply:
You were correct about the `resolve` issue. My `eval` was very heavily flawed, although I wasn't quite sure just how badly. Hence why I was asking for other suggestions. I was not aware that a string could be passed into the Function constructor yesterday. It does perform a lot better and does not have the exact same bug.

Reply:
Thank you.

Reply:
Yeah. This stuff is tricky. I didn’t actually test my hypothesis but it’s fun to hear it worked. 😀

You can pass dependencies as arguments when calling the dynamically generated function. Hopefully you won’t need too many of them.

Best of luck. I definitely think the regex checking leaves a lot to be desired when it comes to allowing Campers to solve the challenges in creative (but still correct) ways.